### PR TITLE
Added index_of and variants thereof

### PR DESCRIPTION
**Pull request by juliusnehring**
`constexpr int64 index_of(string_view s) const;`
`constexpr int64 last_index_of(string_view s) const;`
`constexpr string_indices_of_range all_indices_of(string_view s) const;`

### CODE REVIEW

File: src/clean-core/string_view.hh [hunk 1]
```diff
@@ -343,6 +385,60 @@ private:
...
+        size_t _size;
+        char const* _sdata;
+        size_t _ssize;
+
+        constexpr string_indices_of_range(char const* data, size_t size, char const* sdata, size_t ssize)
+          : _data(data), _size(size), _sdata(sdata), _ssize(ssize)
+        {
+        }
+
+        constexpr auto begin() { return string_indices_of_iterator(_data, _size, _sdata, _ssize); }
```
Comment (Philip-Trettner) [inline src/clean-core/string_view.hh]: the non-const version is not needed

File: src/clean-core/string_view.hh [hunk 2]
```diff
@@ -131,6 +152,27 @@ public:
         return -1;
     }
 
+    /// returns the index of the start of the last occurrence of the character sequence (or -1 if not found)
+    constexpr int64 last_index_of(string_view s) const
+    {
+        if (s.size() > _size || s.empty())
```
Comment (Philip-Trettner) [inline src/clean-core/string_view.hh]: should non-empty be a contract instead?

File: src/clean-core/string_view.hh (hunk 1 above)
Comment (juliusnehring) [reply src/clean-core/string_view.hh]: Good point.

File: src/clean-core/string_view.hh (hunk 2 above)
Comment (juliusnehring) [reply src/clean-core/string_view.hh]: Yeah probably. At least I can't think of a use-case where searching for an empty string is not an error, or at least something that can't be caught on the user side.

**Review by Philip-Trettner — APPROVED** [review]

**Pull request merged**